src/storage-types/src/sources.proto

```diff
@@ -70,6 +70,17 @@ message ProtoCompression {
     }
 }
 
+// NOTE: this message is encoded and stored as part of a source export
+// statement option (currently only `CREATE SUBSOURCE` statements)
+// Be extra careful about changes, ensuring that all changes are backwards compatible
+message ProtoSourceExportStatementDetails {
+    oneof kind {
+        mz_storage_types.sources.postgres.ProtoPostgresSourceExportStatementDetails postgres = 1;
+        mz_storage_types.sources.mysql.ProtoMySqlSourceExportStatementDetails mysql = 2;
+        mz_storage_types.sources.load_generator.ProtoLoadGeneratorSourceExportStatementDetails loadgen = 3;
```

Comment:
Not sure if it's worth defining a new message type. This could just be `google.protobuf.Empty` and then you always instantiate the empty enum variant of the rust type

Reply:
Sounds good! I had mistakenly assumed google.protobuf.Empty wouldn't be wire-compatible with swapping this out for a new message-type in the future, but it apparently is: https://stackoverflow.com/questions/50993815/is-google-protobuf-empty-dangerous-for-backwards-compatibility